packages/optimistic-oracle/src/proposer.js

```diff
@@ -73,60 +73,209 @@ class OptimisticOracleProposer {
+        proposal.estimateGas({ from: this.account })
+      ]);
+    } catch (error) {
+      this.logger.error({
+        at: "OptimisticOracleProposer#sendProposals",
+        message: "Cannot propose price: not enough collateral (or large enough approval)✋",
+        proposer: this.account,
+        proposalBond,
+        priceRequest,
+        error
+      });
+      return;
+    }
+    const txnConfig = {
+      from: this.account,
+      gas: Math.min(Math.floor(gasEstimation * this.GAS_LIMIT_BUFFER), this.txnGasLimit),
+      gasPrice: this.gasEstimator.getCurrentFastPrice()
+    };
+
+    this.logger.debug({
+      at: "OptimisticOracleProposer#sendProposals",
+      message: "Proposing new price",
+      priceRequest,
+      proposalPrice,
+      txnConfig
+    });
+
+    // Send the transaction or report failure.
+    let receipt;
+    try {
+      receipt = await proposal.send(txnConfig);
+    } catch (error) {
+      this.logger.error({
+        at: "OptimisticOracleProposer#sendProposals",
+        message: "Failed to propose price🚨",
+        error
+      });
+      return;
+    }
+    const logResult = {
+      tx: receipt.transactionHash,
+      requester: receipt.events.ProposePrice.returnValues.requester,
+      proposer: receipt.events.ProposePrice.returnValues.proposer,
+      identifier: this.hexToUtf8(receipt.events.ProposePrice.returnValues.identifier),
+      ancillaryData: receipt.events.ProposePrice.returnValues.ancillaryData,
+      timestamp: receipt.events.ProposePrice.returnValues.timestamp,
+      proposedPrice: receipt.events.ProposePrice.returnValues.proposedPrice,
+      expirationTimestamp: receipt.events.ProposePrice.returnValues.expirationTimestamp
+    };
+    this.logger.info({
+      at: "OptimisticOracleProposer#sendProposals",
+      message: "Proposed price!💍",
+      priceRequest,
+      proposalPrice,
+      txnConfig,
+      proposalResult: logResult
+    });
+  }
+  // Construct dispute transaction and send or return early if an error is encountered.
+  async _sendDispute(priceRequest) {
+    // Get proposal price
+    let proposalPrice = priceRequest.proposedPrice;
+
+    // Create pricefeed for identifier
+    const priceFeed = await this._createOrGetCachedPriceFeed(priceRequest.identifier);
+
+    // Pricefeed is either constructed correctly or is null.
+    if (!priceFeed) {
+      this.logger.error({
+        at: "OptimisticOracleProposer#sendDisputes",
+        message: "Failed to construct a PriceFeed for price request",
+        priceRequest
+      });
+      return;
+    }
+
+    // With pricefeed successfully constructed, confirm the proposal price
+    await priceFeed.update();
+    let disputePrice;
+    try {
+      disputePrice = (await priceFeed.getHistoricalPrice(priceRequest.timestamp)).toString();
+    } catch (error) {
+      this.logger.error({
+        at: "OptimisticOracleProposer#sendDisputes",
+        message: "Failed to query historical price for price request",
```

Comment:
nit: might be useful to include the timestamp and certain fields of the price request in this error log.

Reply:
Ah this is a typo, this should print out `priceRequest` not `proposalPrice` which should be undefined if we catch an error here.